src/libs/ck-libs/ampi/ampi.C

```diff
@@ -9364,9 +9364,7 @@ AMPI_API_IMPL(int, MPI_Comm_free, MPI_Comm *comm)
     if (*comm != MPI_COMM_WORLD && *comm != MPI_COMM_SELF) {
       ampi* ptr = getAmpiInstance(*comm);
       ptr->barrier();
```

Comment:
There is no real reason to have a barrier here, and getting rid of it eliminates the occasional failure on linux-smp. I still don't understand the underlying issue with it, but the barrier itself is unnecessary in terms of the MPI semantics. I haven't tested on Windows.